infoschema/builder.go

```diff
@@ -138,6 +138,20 @@ func (b *Builder) applyDropSchema(schemaID int64) []int64 {
 		return nil
 	}
 	delete(b.is.schemaMap, di.Name.L)
+
+	// Copy the sortedTables that contain the table we are going to drop.
+	bucketIdxMap := make(map[int]struct{})
+	for _, tbl := range di.Tables {
+		bucketIdxMap[tableBucketIdx(tbl.ID)] = struct{}{}
+	}
+	buckets := b.is.sortedTablesBuckets
+	for bucketIdx := range bucketIdxMap {
+		oldSortedTables := buckets[bucketIdx]
+		newSortedTables := make(sortedTables, len(oldSortedTables))
+		copy(newSortedTables, oldSortedTables)
+		buckets[bucketIdx] = newSortedTables
```

Comment:
Can we extract this as a function since it also appears at [L106](https://github.com/pingcap/tidb/pull/5672/files#diff-b2feec1c96cf9588e0fd6d139832a071L106) and [L112](https://github.com/pingcap/tidb/pull/5672/files#diff-b2feec1c96cf9588e0fd6d139832a071L112)?